tools/sloths/src/components/memory/GameField.vue

```diff
@@ -0,0 +1,468 @@
+      memorySprite: MEMORY_GAME_SRPITE,
+      cardWinner: MEMORY_GAME_WINNER,
+      images: [] as string[],
+      cards: [] as Card[],
+      activeCard: Infinity,
+      steps: 0,
+      startTime: 0,
+      endTime: 0,
+      grid: '1000px',
+      isHandled: false,
+      isAnimated: false,
+      isModalVisible: false,
+    };
+  },
+
+  props: {
+    level: {
+      type: Object as PropType<MemoryLevel>,
+      default: MEMORY_LEVELS[1],
+    },
+  },
+
+  computed: {
+    ...mapWritableState(themeProp, ['currTheme']),
+
+    getLevel(): string {
+      return `memory.${this.level.level}`;
+    },
+
+    getStepsText(): string {
+      return ruNounEnding(this.steps, this.$t('memory.steps1'), this.$t('memory.steps2'), this.$t('memory.stepsN'));
+    },
+
+    getTime(): number {
+      return this.endTime - this.startTime;
+    },
+
+    getShowModal(): boolean {
+      return this.isModalVisible && !this.isAnimated && !this.isHandled;
+    },
+  },
+
+  mounted() {
+    this.setGrid();
+    this.getImages();
+    this.startGame();
+  },
+
+  watch: {
+    level() {
+      this.setGrid();
+      this.getCards();
+      this.startGame();
+    },
+  },
+
+  methods: {
+    getImages() {
+      this.images = [
+        `${MEMORY_GAME_SRPITE}#memory01`,
+        `${MEMORY_GAME_SRPITE}#memory02`,
+        `${MEMORY_GAME_SRPITE}#memory03`,
+        `${MEMORY_GAME_SRPITE}#memory04`,
+        `${MEMORY_GAME_SRPITE}#memory05`,
+        `${MEMORY_GAME_SRPITE}#memory06`,
+        `${MEMORY_GAME_SRPITE}#memory07`,
+        `${MEMORY_GAME_SRPITE}#memory08`,
+        `${MEMORY_GAME_SRPITE}#memory09`,
+        `${MEMORY_GAME_SRPITE}#memory10`,
+        `${MEMORY_GAME_SRPITE}#memory11`,
+        `${MEMORY_GAME_SRPITE}#memory12`,
+      ];
+
+      this.getCards();
+    },
+
+    getCards() {
+      this.changeScrollHidden('hidden');
+      this.cards = [];
+      let index = 0;
+
+      const images = this.images.sort(() => Math.random() - 0.5).filter((el, i) => i < this.level.n);
+
+      images.forEach((el, i) => {
+        this.cards.push({ img: el, id: i, index, open: false, success: false });
+        index += 1;
+        this.cards.push({ img: el, id: i, index, open: false, success: false });
+        index += 1;
+      });
+      setTimeout(() => this.changeScrollHidden(''), MEMORY_GAME_TIMEOUT);
+    },
+
+    changeScrollHidden(val = '') {
+      const mainEl: HTMLElement | null = document.querySelector('.main');
+      if (mainEl instanceof HTMLElement) {
+        mainEl.style.overflowY = val;
+      }
+    },
+
+    startGame() {
+      const isAllClosed = this.cards.every((el) => !el.open);
+
+      if (!isAllClosed) {
+        playAudio(audioFlip);
+        this.cards.forEach((el, i) => this.closeCard(i));
+      }
+
+      this.activeCard = Infinity;
+      this.steps = 0;
+      this.startTime = 0;
+      this.endTime = 0;
+
+      if (isAllClosed) {
+        playAudio(audioSlide);
+        this.cards.sort(() => Math.random() - 0.5);
```

Comment:
Please consider to avoid duplication for these lines

Reply:
done
add shuffleCards() method